test/test_sync.py

```diff
@@ -554,3 +558,78 @@ def test_sync_without_references(mc, project_name: str, config_update: dict):
     sql = "SELECT count(*) FROM notes WHERE ext_url IS NULL"
     gpkg_cur.execute(sql)
     assert gpkg_cur.fetchone()[0] == 3
+
+
+def test_google_drive_backend(mc):
+    """Test media sync connected to google drive backend"""
+    project_name = "mediasync_test_googledrive"
+    full_project_name = WORKSPACE + "/" + project_name
+    work_project_dir = os.path.join(TMP_DIR, project_name + "_work")
+
+    cleanup(mc, full_project_name, [work_project_dir])
+    prepare_mergin_project(mc, full_project_name)
+
+    # patch config to fit testing purposes
+    config.update(
+        {
+            "MERGIN__USERNAME": API_USER,
+            "MERGIN__PASSWORD": USER_PWD,
+            "MERGIN__URL": SERVER_URL,
+            "MERGIN__PROJECT_NAME": full_project_name,
+            "PROJECT_WORKING_DIR": work_project_dir,
+            "OPERATION_MODE": "copy",
+            "REFERENCES": [
+                {
+                    "file": None,
+                    "table": None,
+                    "local_path_column": None,
+                    "driver_path_column": None,
+                }
+            ],
+            "DRIVER": "google_drive",
+            "GOOGLE_DRIVE__SERVICE_ACCOUNT_FILE": GOOGLE_DRIVE_SERVICE_ACCOUNT_FILE,
+            "GOOGLE_DRIVE__FOLDER": GOOGLE_DRIVE_FOLDER,
+            "GOOGLE_DRIVE__SHARE_WITH": "jan.caha@lutraconsulting.co.uk",
+        }
+    )
+
+    google_drive_delete_folder(GoogleDriveDriver(config), GOOGLE_DRIVE_FOLDER)
+
+    main()
+
+    google_drive_files = google_drive_list_files_in_folder(
+        GoogleDriveDriver(config), GOOGLE_DRIVE_FOLDER
+    )
+
+    assert "img1.png" in google_drive_files
+    assert "images/img2.jpg" in google_drive_files
+
+    # files in mergin project still exist (copy mode)
+    assert os.path.exists(os.path.join(work_project_dir, "img1.png"))
+    assert os.path.exists(os.path.join(work_project_dir, "images", "img2.jpg"))
+
+    # test for moving files
+    cleanup(mc, full_project_name, [work_project_dir])
+    prepare_mergin_project(mc, full_project_name)
+
+    # patch config to fit testing purposes
+    config.update(
+        {
+            "OPERATION_MODE": "move",
```

Comment:
Do we need to test both, move and copy modes, is not is driver agnostic?

Reply:
yes, it is. It was for my testing mostly and i left it there...removed